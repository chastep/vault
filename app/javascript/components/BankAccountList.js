import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Header, Button, List } from "semantic-ui-react";

class BankAccountList extends Component {
  renderBankAccounts() {
    const { bankAccounts } = this.props;
    bankAccounts.sort(
      (a, b) => new Date(b.created_at) - new Date(a.created_at),
    );

    return bankAccounts.map(bankAcct => (
      <li key={bankAcct.id}>
        <Link to={`/bank_accounts/${bankAcct.id}`}>
          {bankAcct.nickname + ' - ' + bankAcct.created_at}
        </Link>
      </li>
    ));
  }

  render() {
    return (
      <section>
        <h2>
          Bank Accounts
          <Link to="/bank_accounts/new">New Bank Account</Link>
        </h2>
        <ul>{this.renderBankAccounts()}</ul>
      </section>
    );
  }
}

BankAccountList.propTypes = {
  bankAccounts: PropTypes.arrayOf(PropTypes.object),
};

BankAccountList.defaultProps = {
  bankAccounts: [],
};

export default BankAccountList;
