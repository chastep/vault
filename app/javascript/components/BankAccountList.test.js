import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'

import BankAccountList from './BankAccountList';

it('renders with no bank accounts', () => {
  const bankAccounts = [];
  const activeId = '';

  const component = renderer.create(
    <MemoryRouter>
      <BankAccountList bankAccounts={bankAccounts} activeId={activeId} />
    </MemoryRouter>
  ).toJSON();
  expect.assertions(1);
  expect(component).toMatchSnapshot();
});

it('renders with bank accounts, none selected', () => {
  const bankAcct = {
    id: 1,
    account_number: '1231231230',
    routing_number: '011000015',
    nickname: 'Test Account',
    location_id: 1,
    address: '123 Fake St',
    address2: 'Apt 4',
    city: 'Chicago',
    region: 'IL',
    postal: '60606',
    created_at: '2019-03-26T18:10:38.029Z',
    updated_at: '2019-03-26T18:10:38.029Z'
  };
  const bankAccounts = [bankAcct];
  const activeId = '';

  const component = renderer.create(
    <MemoryRouter>
      <BankAccountList bankAccounts={bankAccounts} activeId={activeId} />
    </MemoryRouter>
  ).toJSON();
  expect.assertions(1);
  expect(component).toMatchSnapshot();
});

it('renders with bank accounts, account selected', () => {
  const bankAcct = {
    id: 1,
    account_number: '1231231230',
    routing_number: '011000015',
    nickname: 'Test Account',
    location_id: 1,
    address: '123 Fake St',
    address2: 'Apt 4',
    city: 'Chicago',
    region: 'IL',
    postal: '60606',
    created_at: '2019-03-26T18:10:38.029Z',
    updated_at: '2019-03-26T18:10:38.029Z'
  };
  const bankAccounts = [bankAcct];
  const activeId = 1;

  const component = renderer.create(
    <MemoryRouter>
      <BankAccountList bankAccounts={bankAccounts} activeId={activeId} />
    </MemoryRouter>
  ).toJSON();
  expect.assertions(1);
  expect(component).toMatchSnapshot();
});
