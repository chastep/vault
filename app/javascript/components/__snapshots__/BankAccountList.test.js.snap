// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with bank accounts, account selected 1`] = `
<section>
  <h2
    className="ui left floated header"
  >
    Bank Accounts
  </h2>
  <h3
    className="ui right floated header"
  >
    <a
      href="/bank_accounts/new"
      onClick={[Function]}
    >
      New
    </a>
  </h3>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Nickname
        </th>
        <th
          className=""
        >
          Account Number
        </th>
        <th
          className=""
        >
          Routing Number
        </th>
        <th
          className=""
        >
          Address
        </th>
        <th
          className=""
        >
          Created At
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className="single line"
        >
          <a
            href="/bank_accounts/1"
            onClick={[Function]}
          >
            Test Account
          </a>
        </td>
        <td
          className=""
        >
          1231231230
        </td>
        <td
          className=""
        >
          011000015
        </td>
        <td
          className=""
        >
          123 Fake St, Apt 4 Chicago, IL 60606
        </td>
        <td
          className="single line"
        >
          03/26/2019 01:10 PM
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`renders with bank accounts, none selected 1`] = `
<section>
  <h2
    className="ui left floated header"
  >
    Bank Accounts
  </h2>
  <h3
    className="ui right floated header"
  >
    <a
      href="/bank_accounts/new"
      onClick={[Function]}
    >
      New
    </a>
  </h3>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Nickname
        </th>
        <th
          className=""
        >
          Account Number
        </th>
        <th
          className=""
        >
          Routing Number
        </th>
        <th
          className=""
        >
          Address
        </th>
        <th
          className=""
        >
          Created At
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className="single line"
        >
          <a
            href="/bank_accounts/1"
            onClick={[Function]}
          >
            Test Account
          </a>
        </td>
        <td
          className=""
        >
          1231231230
        </td>
        <td
          className=""
        >
          011000015
        </td>
        <td
          className=""
        >
          123 Fake St, Apt 4 Chicago, IL 60606
        </td>
        <td
          className="single line"
        >
          03/26/2019 01:10 PM
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`renders with no bank accounts 1`] = `
<section>
  <h2
    className="ui left floated header"
  >
    Bank Accounts
  </h2>
  <h3
    className="ui right floated header"
  >
    <a
      href="/bank_accounts/new"
      onClick={[Function]}
    >
      New
    </a>
  </h3>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Nickname
        </th>
        <th
          className=""
        >
          Account Number
        </th>
        <th
          className=""
        >
          Routing Number
        </th>
        <th
          className=""
        >
          Address
        </th>
        <th
          className=""
        >
          Created At
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    />
  </table>
</section>
`;
